ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1459,6 +1464,22 @@ public Table getTable(final String dbName, final String tableName, boolean throw
     return new Table(tTable);
   }
 
+  /**
+   * Get ValidWriteIdList for the current transaction.
+   * @param dbName
+   * @param tableName
+   * @return
+   * @throws LockException
+   */
+  private ValidWriteIdList getValidWriteIdList(String dbName, String tableName) throws LockException {
+    ValidWriteIdList validWriteIdList = null;
+    long txnId = SessionState.get().getTxnMgr() != null ? SessionState.get().getTxnMgr().getCurrentTxnId() : 0;
+    if (txnId > 0) {
```

Comment:
I think we should we explicitly checking conf.get(ValidTxnList.VALID_TXNS_KEY) != null here instead of txnId > 0 since txnId > 0 doesn't guarantee that validTransactions have been recorded in the conf. On a side note, its bit weird DbTxnManager code using assert statements instead of Preconditions since jvm disables assert statements by default. So for example, if a given code-path doesn't have VALID_TXNS_KEY set in the conf, this method will do and fetch the current ValidWriteIdList for the table which is not the snapshot view of the table during query compilation.

Reply:
Thats taken care in getTableValidWriteIdListWithTxnList() method. It fails, if the valid_txns_key is not set. Thats why I had to explicitly set this    conf.set(ValidTxnList.VALID_TXNS_KEY,
        new ValidReadTxnList(new long[0], new BitSet(), 1000, Long.MAX_VALUE).writeToString()); in one of the tests, which doesn't go through Driver flow, but invokes SemanticAnalyzer directly for a transactional table.